Test/civl/inductive-sequentialization/snapshot-scatter-gather.bpl

```diff
@@ -0,0 +1,168 @@
+type Value;
+
+type Tid;
+
+datatype StampedValue {
+  StampedValue(ts: int, value: Value)
+} 
+
+const n: int;
+axiom n >= 1;
+
+var {:layer 0,2} mem: [int]StampedValue;
+var {:layer 0,2} r1: [Tid][int]StampedValue;  
+var {:layer 0,2} gather1: [Tid][int]bool;
```

Comment:
I find the use of ```gather1``` and ```gather2``` odd. Can we replace them with the use of async calls? Something  like this:

- main_f splits tid into n pieces tid_1, ..., tid_n, and creates the pending asyncs, where the i-th pending async is read_f(tid_i, i)
- each pending async read_f(tid_i, i) reads mem[i] into a local value v and creates a pending async read_resp_f(tid_i, i, v)
- read_resp_f(tid_i, i, v) updates r1[i] with v